docs/CONTRIBUTING.md

```diff
@@ -1,21 +1,271 @@
-* Go to [open PRs](https://github.com/web-platform-dx/web-features/pulls) to review incoming feature groups and updates.
+In any case, **thank you** for wanting to help. This document will guide you through the process of contributing to the web-features repository.
 
-Also consider joining [the WebDX Community Group](https://www.w3.org/community/webdx/).
+Also consider joining the [WebDX Community Group](https://www.w3.org/community/webdx/).
+
+## What makes a feature
+
+Think of features as individual components of the web platform that web developers can use to achieve specific tasks.
+
+As such, a feature has no specific size. Some features might cover single CSS properties while other features cover entire JavaScript APIs with multiple interfaces, methods, and properties. As long as a web developer would use this to achieve a specific goal, it's a feature. 
+
+For example, the `fetch` API is a feature, the `:has()` CSS pseudo-class function too, and so is the Web Audio API.
+
+### Feature file name and format
+
+Features in this repository are authored as **YAML files** that are stored in the [`features`](https://github.com/web-platform-dx/web-features/tree/main/features) directory. Each file corresponds to a single feature, and contains metadata about the feature, such as its name, description, specification, or support data.
+
+For example, the `fetch` API feature is described in the [`features/fetch.yml`](https://github.com/web-platform-dx/web-features/blob/main/features/fetch.yml) file.
+
+The name of the file also constitutes the unique ID of the feature. In the example above, the unique ID of the feature is `fetch`.
+
+Feature IDs must be unique as they are used in other projects to reference the feature. For example, the [browser-compat-data](https://github.com/mdn/browser-compat-data/) project references feature IDs in their compatibility data.
+
+### Fields in a feature file
+
+This table lists the fields that can be found in a feature file, and provides a brief description of each field:
+
+| Field | Description | Type | Mandatory |
+|---|---|---|---|
+| `name` | The name of the feature. | String | Yes |
+| `alias` | An alternative name for the feature. | String, or array of strings | No |
+| `description` | A short description of the feature. | String | Yes |
+| `spec` | One or more specification URLs for this feature | String, or array of strings | Yes |
+| `group` | An optional group, or list of groups that this feature belongs to | String, or array of strings | No |
+| `snapshot` | An optional snapshot, or list of snapshots that this feature belongs to | String, or array of strings | No |
+| `caniuse` | The feature's ID on the [Can I Use](https://caniuse.com/) website. | String | No |
+| `compat_features` | The list of browser-compat-data entries that make up this feature. In many cases, the corresponding browser-compat-data entries already map to the feature ID, and this is therefore not needed. | Array of strings | No |
```

Comment:
Very good questions, @tidoust!

`compat_features` is the easiest way to author a new feature, but then we want tags in BCD for the long term, so that web-features doesn't have to react to renames and the like.

I have a script in https://github.com/web-platform-dx/web-features/pull/473 to migrate tags, but I've only run it 2 or 3 times, there's no schedule.

A situation which is likely to get contributors confused is when BCD has tags but you need to add or remove something. Then you need to list all of the keys that should be included in `compat_features`, completely ignoring BCD. But how do you get back in sync with BCD? The current answer is by hand, with some pain.

Reply:
This goes back to your general comment @tidoust, stating that it might make sense to almost have 2 contribution guides (or 2 sections). One where you focus on creating the feature at all, with a good ID, name, description. And one, more involved, where you spend time identifying the right BCD keys and which ones the status should be computed from.
I'm not against that. I like beginner-friendly guides that help newcomers get from nothing to a PR.
Since this is a larger piece of work, how about we do this in a separate PR?